zipline/pipeline/domain.py

```diff
@@ -71,6 +71,32 @@ def data_query_cutoff_for_sessions(self, sessions):
             "available" on each session.
         """
 
+    @default
+    def roll_forward(self, dt):
+        """
+        Given a date, align it to the calendar of the pipeline's domain.
+
+        Parameters
+        ----------
+        dt : pd.Timestamp
+
+        Returns
+        -------
+        pd.Timestamp
+        """
+        dt = pd.Timestamp(dt, tz='UTC')
+
+        trading_days = self.all_sessions()
+        if dt in trading_days:
+            return dt
+        try:
+            return trading_days[trading_days.searchsorted(dt)]
+        except IndexError:
+            raise IndexError(
```

Comment:
This should be a ValueError. From the caller's perspective, an IndexError is a strange error, because they're not interacting with an array/sequence.